### ISSUE TITLE
catch errors thrown by image-size

### ISSUE THREAD

**Issue by billiegoose** [elingerojo/contexter-cli#2]
Apparently this can happen:

```
E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\types\svg.js:72
  throw new TypeError('invalid svg');
  ^

TypeError: invalid svg
    at Object.calculate (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\types\svg.js:72:9)
    at lookup (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\index.js:26:31)
    at module.exports (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\index.js:100:12)
    at File.setDimensions (E:\global_modules\node_modules\contexter-cli\lib\files\image.js:23:23)
    at Object.parse (E:\global_modules\node_modules\contexter-cli\lib\plugins\app-image.js:29:10)
    at File.getContent (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:167:24)
    at File.squeeze (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:146:10)
    at File.initialize (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:36:10)
    at Context.newFile (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\context.js:51:13)
    at createFile (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\index.js:85:15)
```

I think it was an SVG font it tripped on... it had just loaded `/doc/fonts/glyphicons-halflings-regular.eot added` so I don't know if that means it crashed on that file or the one immediately afterward.

Just throw a try/catch wrapper around that and you should be good!

Edit: yup, crash is definitely from the font `glyphicons-halflings-regular.svg`

**Comment by billiegoose** [comment]
Not just SVGs. Apparently `image-size` can explode on JPEG as well

```
E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\types\jpg.js:24
    throw new TypeError('Corrupt JPG, exceeded buffer limits');
    ^

TypeError: Corrupt JPG, exceeded buffer limits
    at validateBuffer (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\types\jpg.js:24:11)
    at Object.calculate (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\types\jpg.js:43:5)
    at lookup (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\index.js:26:31)
    at module.exports (E:\global_modules\node_modules\contexter-cli\node_modules\image-size\lib\index.js:100:12)
    at File.setDimensions (E:\global_modules\node_modules\contexter-cli\lib\files\image.js:23:23)
    at Object.parse (E:\global_modules\node_modules\contexter-cli\lib\plugins\app-image.js:29:10)
    at File.getContent (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:167:24)
    at File.squeeze (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:146:10)
    at File.initialize (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\file.js:36:10)
    at Context.newFile (E:\global_modules\node_modules\contexter-cli\node_modules\contexter\lib\context.js:51:13)
```

**Comment by elingerojo** [comment]
Ok,
Let me take a look

**Comment by billiegoose** [comment]
I remember using `image-size` in a mission-critical project, and just looked up the code. Everywhere I used it I wrapped it in try/catch. Not sure what to do if it fails... in my project I was looking for images of a particular size so I just considered errors as "no match" but in this case I'm not sure. Maybe you'd want to report that, dunno.

Anyway I'm really excited, and will keep trying `contexter-cli serve .` on directories until I find one that works! 😆

**Comment by elingerojo** [comment]
It is only used in one place. So the try/catch is simple.

I am working on it and will push it right away.

**Comment by billiegoose** [comment]
Eee! It works on my website dir! (after I removed the node_modules folder... funny how much stuff can accumulate in there)

![image](https://cloud.githubusercontent.com/assets/587740/25292683/35708b1a-26a5-11e7-8419-5f1761dbe656.png)

I'm already thinking of all sorts of plugin ideas - parse CNAME! match LICENSE to spdx identifier! parse .md! - and all the things you could build with this. Because this is like "the thing you would use to build a build tool". You could use it to power something like Webpack (or Jus) or even a compiler like Babel, or even a desktop search engine.

**Comment by elingerojo** [comment]
Just bumped version 0.1.4

**Comment by elingerojo** [comment]
I am working on adding a way "ignore" stuff

Will let a CLI parameter to set a .gitignore-like file (it could be the real .gitignore file)

Something like:

```
$ contexter-cli serve . --ignores "./.gitignore"
```

...but also I am thinking of using an opposite approach.

Instead of defining 'ignores', construct 'globs' automatically from the plugins.

This way, everything is 'ignored' unless it is in a plugin `check()`

...then, the user can skip the CLI --ignores parameter if wanted

**Comment by elingerojo** [comment]
> this is like "the thing you would use to build a build tool". You could use it to power something like Webpack (or Jus) or even a compiler like Babel, or even a desktop search engine.

Yeah!
> "filesystem watcher + configurable virtual files + super parser powers" is dynamite!

😃

**Comment by billiegoose** [comment]
Eh... don't add too many features to contexter-cli if it's just meant to serve as an example application. I wouldn't worry about ignoring directories... maybe hard-code an assumption to ignore node_modules but I'm lazy.

0.1.4 still crashes. You do catch the error... but you're rethrowing it again! 😆 Maybe do

```
setDimensions () {
  try {
    this.dimensions = imageSize(this.path.processRelative)
  } catch (e) {
    this.dimensions = null
  }
}
```

of course that would mean dealing with the `null` somewhere else down the line.

**Comment by elingerojo** [comment]
Done!
...bump to version 0.1.5

**Comment by billiegoose** [comment]
And it works! No crash now.
![image](https://cloud.githubusercontent.com/assets/587740/25295794/22a426be-26b3-11e7-9326-674a18bedaac.png)

**Comment by elingerojo** [comment]
Found the root problem.

`app-image.js` plugin `check()` had a bug 😰 

I was already 'ignoring' `'svg'` fonts masked as images but the code was not working because I forgot the dot in the extension

```javascript
    if (extension === '.svg') {
      isSvgFont = fs.readFileSync(filename, 'utf8').indexOf('</font>') > -1
    }
```

Corrected but left the try/catch anyway.

Now, will not parse the file as images so not `dimensions: null` anymore (...well, maybe for other cases where `npm image-size` truly fails).

The file will appear as `unknown` as it should be

Included a CI/test and bumped to version 0.1.6
